library/core/src/intrinsics/bounds.rs

```diff
@@ -0,0 +1,36 @@
+//! Various traits used to restrict intrinsics to not-completely-wrong types.
+
+/// Types with a built-in dereference operator in runtime MIR,
+/// aka references and raw pointers.
+pub trait BuiltinDeref: Sized {
```

Comment:
kinda feels obvious but
```suggestion
pub unsafe trait BuiltinDeref: Sized {
```